ddl/partition.go

```diff
@@ -1664,31 +1664,13 @@ func checkPartitionExprAllowed(_ sessionctx.Context, _ *model.TableInfo, e ast.E
 			return errors.Trace(ErrPartitionFunctionIsNotAllowed)
 		}
 	case *ast.UnaryOperationExpr:
-		if v.Op == opcode.BitNeg {
+		switch v.Op {
+		case opcode.BitNeg, opcode.Not, opcode.Not2:
 			return errors.Trace(ErrPartitionFunctionIsNotAllowed)
 		}
 	}
 	return nil
 }
-
-func checkPartitionExprFuncAllowed(_ sessionctx.Context, _ *model.TableInfo, e ast.ExprNode) error {
```

Comment:
this check is needed because only these functions are allowed to be partition expression. please add them to function `checkPartitionExprAllowed`